iele.md

```diff
@@ -406,6 +406,129 @@ Some instructions require an argument to be interpreted as an address (modulo 16
     rule #addr?(REGS1 = staticcall LABEL at W0 (REGS2) gaslimit W1)         => REGS1 = staticcall LABEL at #addr(W0) (REGS2) gaslimit W1
     rule #addr?(OP)                                                         => OP [owise]
+```
+
+### Internal Operations
+
+-   `#loadAccount_` allows declaring a new empty account with the given address (and assumes the rounding to 160 bits has already occured).
+    If the account already exists with non-zero nonce or non-empty code, an exception is thrown.
+    Otherwise, if the account already exists, the storage is cleared.
+
+```{.k .uiuck .rvk .standalone .node}
+    syntax InternalOp ::= "#loadAccount" Int
+                        | "#initAccount" Int
+ // ----------------------------------------
+    rule <k> #initAccount ACCT => #exception ACCT_COLLISION ... </k>
+         <account>
+           <acctID> ACCT  </acctID>
+           <code>   CODE  </code>
+           <nonce>  NONCE </nonce>
+           ...
+         </account>
+      requires CODE =/=K #emptyCode orBool NONCE =/=K 0
+
+    rule <k> #initAccount ACCT => . ... </k>
+         <account>
+           <acctID>  ACCT       </acctID>
+           <code>    #emptyCode </code>
+           <nonce>   0          </nonce>
+           <storage> _ => .Map  </storage>
+           ...
+         </account>
+
+    rule <k> (.K => #loadAccount ACCT) ~> #initAccount ACCT ... </k>
+         <activeAccounts> ACCTS </activeAccounts>
+      requires notBool ACCT in ACCTS
+```
+
+```{.k .uiuck .rvk .standalone}
+    rule <k> #loadAccount ACCT => . ... </k>
+         <activeAccounts> ACCTS (.Set => SetItem(ACCT)) </activeAccounts>
+         <accounts>
+           ( .Bag
+          => <account>
+               <acctID>    ACCT       </acctID>
+               ...
+             </account>
+           )
+           ...
+         </accounts>
+      requires notBool ACCT in ACCTS
+```
+
+-   `#lookupStorage` loads the `<storage>` cell with the storage value for a particular key in the account.
+
+```{.k .uiuck .rvk .standalone .node}
+    syntax InternalOp ::= #lookupStorage ( Int , Int )
+ // --------------------------------------------------
+    rule <k> #lookupStorage(ACCT, INDEX) => . ... </k>
+         <account>
+           <acctID>  ACCT    </acctID>
+           <storage> STORAGE </storage>
+           ...
+         </account>
+      requires INDEX in_keys(STORAGE)
+```
+
+```{.k .uiuck .rvk .standalone}
+    rule <k> #lookupStorage(ACCT, INDEX) => . ... </k>
+         <account>
+           <acctID>  ACCT                              </acctID>
+           <storage> STORAGE => STORAGE [ INDEX <- 0 ] </storage>
+           ...
+         </account>
+      requires notBool INDEX in_keys(STORAGE)
+```
+
+-   `#lookupCode` loads the `<code>` cell with the code for a particular account.
+
+```{.k .uiuck .rvk .standalone .node}
+    syntax InternalOp ::= #lookupCode ( Int )
```

Comment:
where are the rewrite rules for this in case of standalone?

Reply:
They are not necessary as lookupCode only applies if the code is .Contract, which is not a well formed contract and is therefore not actually ever the code of any account; we simply reserve it to refer to an account that the node has not yet lazily computed code for.